src/StackExchange.Exceptional.MongoDB/MongoDBErrorStore.cs

```diff
@@ -0,0 +1,295 @@
+        private readonly int _displayCount;
+        private readonly string _connectionString;
+
+        private static bool classMapRegistered;
+
+        /// <summary>
+        /// The maximum count of errors to show.
+        /// </summary>
+        public const int MaximumDisplayCount = 500;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="MongoDBErrorStore" /> with the specified connection string.
+        /// The default table name is "Exceptions".
+        /// </summary>
+        /// <param name="connectionString">The database connection string to use.</param>
+        /// <param name="applicationName">The application name to use when logging.</param>
+        public MongoDBErrorStore(string connectionString, string applicationName)
+            : this(new ErrorStoreSettings()
+            {
+                ApplicationName = applicationName,
+                ConnectionString = connectionString
+            })
+        { }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="ErrorStoreSettings"/> with the given configuration.
+        /// The default table name is "Exceptions".
+        /// </summary>
+        /// <param name="settings">The <see cref="ErrorStoreSettings"/> for this store.</param>
+        public MongoDBErrorStore(ErrorStoreSettings settings) : base(settings)
+        {
+            RegisterClassMap();
+
+            _displayCount = Math.Min(settings.Size, MaximumDisplayCount);
+            _connectionString = settings.ConnectionString;
+            _tableName = settings.TableName ?? "Exceptions";
+
+            if (_connectionString.IsNullOrEmpty())
+                throw new ArgumentOutOfRangeException(nameof(settings), "A connection string or connection string name must be specified when using a MongoDB error store");
+        }
+
+        private static void RegisterClassMap()
+        {
+            if (classMapRegistered) { return; }
+            BsonClassMap.RegisterClassMap<Error>(cm =>
+            {
+                cm.AutoMap();
+                cm.MapIdMember(x => x.GUID);
+                foreach (var prop in typeof(Error).GetMembers())
+                {
+                    if (prop.CustomAttributes.Any(x => x.AttributeType == typeof(JsonIgnoreAttribute)))
+                    {
+                        cm.UnmapMember(prop);
+                    }
+                }
+            });
+            classMapRegistered = true;
+        }
+
+        /// <summary>
+        /// Protects an error from deletion, by making IsProtected = true in the database.
+        /// </summary>
+        /// <param name="guid">The GUID of the error to protect.</param>
+        /// <returns><c>true</c> if the error was found and protected, <c>false</c> otherwise.</returns>
+        protected override async Task<bool> ProtectErrorAsync(Guid guid)
+        {
+            var c = GetConnection();
+            var builders = Builders<Error>.Filter;
```

Comment:
Is this a pattern I'm not familiar with? It appears `builders` isn't used anywhere, but is it required as a call? (same for most methods here)